engine/src/test/scala/engine/workflow/ValidateActorSpec.scala

```diff
@@ -0,0 +1,177 @@
+package engine.workflow
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.testkit.{DefaultTimeout, ImplicitSender, TestKit}
+import cromwell.engine.backend.{BackendType, Backend}
+import cromwell.engine.workflow.ValidateActor.{ValidateWorkflow, ValidationFailure, ValidationSuccess}
+import cromwell.engine.workflow.{ValidateActor, WorkflowOptions}
+import cromwell.util.SampleWdl.HelloWorld
+import org.mockito.Mockito._
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{BeforeAndAfter, Matchers, WordSpecLike}
+import spray.json.DefaultJsonProtocol._
+import spray.json._
+import wdl4s.ValidationException
+
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+class ValidateActorSpec extends TestKit(ActorSystem("ValidateActorSpecSystem"))
+  with DefaultTimeout with ImplicitSender with WordSpecLike with Matchers with BeforeAndAfter with MockitoSugar {
+  val MissingInputsJson: String = "{}"
+  val MalformedJson: String = "foobar bad json!"
+  val MalformedWdl: String = "foobar bad wdl!"
+
+  val validOptionsJson = HelloWorld.rawInputs.toJson.toString()
+  val malformedOptionsJson = MalformedJson
+  val validInputsJson = HelloWorld.rawInputs.toJson.toString()
+  val malformedInputsJson = MalformedJson
+  val invalidInputJson = Map("llo.addsee" -> "wod").toJson.toString()
+  val validRunAttr =
+    """
+      |runtime {
+      |  docker: "ubuntu:latest"
+      |}
+    """.stripMargin
+  val invalidRunAttr =
+    """
+      |runtime {
+      |  asda: "sda"
+      |}
+    """.stripMargin
+  val validWdlSource = HelloWorld.wdlSource(validRunAttr)
+  val invalidWdlSource = HelloWorld.wdlSource(invalidRunAttr)
+
+  val backendMock = mock[Backend]
+
+  var validateActor: ActorRef = _
+
+  before {
+    validateActor = system.actorOf(ValidateActor.props(backendMock))
+  }
+
+  after {
+    system.stop(validateActor)
+  }
+
+  "A Validate actor" should {
+    "return ValidationSuccess when Namespace, options, runtime attributes and inputs are valid" in {
+      within(1000 millis) {
```

Comment:
I'd pull this out to a constant, reduce, and dilate for pokey Travis builds that might need more time:

```scala
val Timeout = 1.second.dilated
```